internal/proxysql/core.go

```diff
@@ -1,160 +1,245 @@
-	err = os.WriteFile(checksumFile, []byte(digest), 0o600)
-	if err != nil {
-		slog.Error("Failed to write to checksum file", slog.String("file", checksumFile), slog.Any("error", err))
+	// Pod is shutting down. Only run this for core pods, as satellites don't need special considerations when
+	// they leave the cluster.
+	if oldpod.Status.Phase == "Running" && newpod.Status.Phase == "Failed" {
+		err := p.removePodFromCluster(oldpod)
+		if err != nil {
+			slog.Error("Error in removePod()", slog.Any("err", err))
+		}
 	}
-
-	slog.Info("Commands ran", slog.String("commands", strings.Join(commands, "; ")))
 }
 
-func GetCorePods(settings *configuration.Config) ([]PodInfo, error) {
-	app := settings.Core.PodSelector.App
-	component := settings.Core.PodSelector.Component
-	namespace := settings.Core.PodSelector.Namespace
-
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, err
-	}
-
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
+// Add the new pod to the cluster.
+//   - If it's a core pod, add it to the proxysql_servers table
+//   - if it's a satellite pod, run the commands to accept it to the cluster
+func (p *ProxySQL) addPodToCluster(pod *v1.Pod) error {
+	slog.Info("Pod joined the cluster", slog.String("name", pod.Name), slog.String("ip", pod.Status.PodIP))
 
-	pods, _ := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("app=%s,component=%s", app, component),
-	})
+	commands := [][]any{}
 
-	var corePods []PodInfo
-	for _, pod := range pods.Items {
-		corePods = append(corePods, PodInfo{PodIP: pod.Status.PodIP, Hostname: pod.Name, UID: string(pod.GetUID())})
+	// If the new pod is a core pod, delete the default entries in the proxysql_server list and add the new pod to it.
+	if pod.Labels["component"] == "core" {
+		// TODO: maybe make this configurable, not everyone will name the service this.
+		commands = append(commands, []any{"DELETE FROM proxysql_servers WHERE hostname = 'proxysql-core'"})
+		commands = append(commands, []any{"INSERT INTO proxysql_servers VALUES (?, 6032, 0, ?)", pod.Status.PodIP, pod.Name})
 	}
 
-	return corePods, err
-}
-
-func calculateChecksum(pods []PodInfo) string {
-	data := []string{}
+	commands = append(commands,
+		[]any{"LOAD PROXYSQL SERVERS TO RUNTIME"},
+		[]any{"LOAD ADMIN VARIABLES TO RUNTIME"},
+		[]any{"LOAD MYSQL VARIABLES TO RUNTIME"},
+		[]any{"LOAD MYSQL SERVERS TO RUNTIME"},
+		[]any{"LOAD MYSQL USERS TO RUNTIME"},
+		[]any{"LOAD MYSQL QUERY RULES TO RUNTIME"},
+	)
 
-	for _, pod := range pods {
-		data = append(data, fmt.Sprintf("%s:%s:%s", pod.PodIP, pod.Hostname, pod.UID))
+	for _, command := range commands {
+		_, err := p.conn.Exec(command[0].(string), command[1:]...)
+		if err != nil {
+			// FIXME: wrap error with extra info and return
+			slog.Error("Commands failed", slog.Any("commands", command), slog.Any("error", err))
```

Comment:
Since this is just one of the commands failing, should this be `Command failed`?